openwisp_controller/config/tests/test_api.py

```diff
@@ -510,3 +512,44 @@ def test_vpn_delete_api(self):
         r = self.client.delete(path)
         self.assertEqual(r.status_code, 204)
         self.assertEqual(Vpn.objects.count(), 0)
+
+    def test_get_request_with_change_perm(self):
+        change_perm = Permission.objects.filter(codename='change_template')
+        user = self._get_user()
+        user.user_permissions.add(*change_perm)
+        org1 = self._get_org()
+        OrganizationUser.objects.create(user=user, organization=org1, is_admin=True)
+        self.client.force_login(user)
+        t1 = self._create_template(name='t1', organization=self._get_org())
+        with self.subTest('Get Template List'):
+            path = reverse('config_api:template_list')
+            response = self.client.get(path)
+            self.assertEqual(response.status_code, 200)
+        with self.subTest('Get Template Detail'):
+            path = reverse('config_api:template_detail', args=[t1.pk])
+            response = self.client.get(path)
+            self.assertEqual(response.status_code, 200)
+
+    def test_get_request_with_view_perm(self):
+        view_perm = Permission.objects.filter(codename='view_template')
+        user = self._get_user()
+        user.user_permissions.add(*view_perm)
+        org1 = self._get_org()
+        OrganizationUser.objects.create(user=user, organization=org1, is_admin=True)
+        self.client.force_login(user)
+        t1 = self._create_template(name='t1', organization=self._get_org())
+        with self.subTest('Get Template List'):
+            path = reverse('config_api:template_list')
+            response = self.client.get(path)
+            self.assertEqual(response.status_code, 200)
+        with self.subTest('Get Template Detail'):
+            path = reverse('config_api:template_detail', args=[t1.pk])
+            response = self.client.get(path)
+            self.assertEqual(response.status_code, 200)
+
+    def test_get_request_with_no_perm(self):
```

Comment:
This is the most important test. The one which fails without the fix.
Please @ManishShah120, whenever you add new features or fix bugs, write tests that fail without your changes, especially if we're dealing with anything related to security like permission, authentication and so on.
Please do this without needing us to tell you to do it, it's like A, B, C, you need to know this and do it automatically, **always**.